libsolutil/JSON.cpp

```diff
@@ -22,135 +22,218 @@
 
 #include <libsolutil/JSON.h>
 
-#include <libsolutil/CommonIO.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/trim.hpp>
 
-#include <boost/algorithm/string/replace.hpp>
-
-#include <sstream>
 #include <map>
-#include <memory>
+#include <sstream>
 
 static_assert(
-	(JSONCPP_VERSION_MAJOR == 1) && (JSONCPP_VERSION_MINOR == 9) && (JSONCPP_VERSION_PATCH == 3),
-	"Unexpected jsoncpp version: " JSONCPP_VERSION_STRING ". Expecting 1.9.3."
-);
+	(NLOHMANN_JSON_VERSION_MAJOR == 3) && (NLOHMANN_JSON_VERSION_MINOR == 11) && (NLOHMANN_JSON_VERSION_PATCH == 3),
+	"Unexpected nlohmann-json version. Expecting 3.11.3.");
 
 namespace solidity::util
 {
 
 namespace
 {
 
-/// StreamWriterBuilder that can be constructed with specific settings
-class StreamWriterBuilder: public Json::StreamWriterBuilder
+/// Takes a JSON value (@ _json) and removes all its members with value 'null' recursively.
+void removeNullMembersHelper(Json& _json)
```

Comment:
This is of course horrible, but I'll pretend I didn't see it.

Reply:
This entire file is a bit horrible :-). But yeah, I may need to have another look at the escaping stuff, but it'll probably be fine.

Reply:
yeah the diff is very confusing.. however, the file itself might be easier readable by viewing the file not as a diff